flake.nix

```
@@ -1,113 +1,99 @@
+          rust-toolchain = (pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml).override {
+              extensions = [ "rust-src" "rust-analyzer" ];
+          };
+
+          runtimeDeps = with pkgs; if stdenv.isDarwin then [
             darwin.libobjc
             darwin.apple_sdk_11_0.frameworks.AppKit
             darwin.apple_sdk_11_0.frameworks.AVFoundation
             darwin.apple_sdk_11_0.frameworks.Vision
-          ] else [
+          ] else (with pkgs; [
+            (lib.getLib gcc-unwrapped)
             fontconfig
             libGL
             libxkbcommon
             vulkan-loader
-          ] ++ (with pkgs; [
-            xorg.libX11
-            xorg.libXcursor
-            xorg.libXi
-            xorg.libXrandr
-            xorg.libxcb
             wayland
+          ]) ++ (with pkgs.xorg; [
+            libX11
+            libXcursor
+            libXi
+            libXrandr
+            libxcb
           ]);
-        in
-        rustPlatform.buildRustPackage {
-          pname = "rio";
-          name = "rio"; # attribute name for packages
-          src = ./.;
-          nativeBuildInputs = with pkgs; [
+
+          buildDeps = with pkgs; [
             ncurses
-            cmake
+            ] ++ lib.optionals stdenv.isLinux [
             pkg-config
+            cmake
             autoPatchelfHook
           ];
 
-          runtimeDependencies = rlinkLibs;
-          buildInputs = rlinkLibs;
-          cargoLock.lockFile = ./Cargo.lock;
+          rustPackage = rust-toolchain: pkgs.rustPlatform.buildRustPackage {
+              inherit (cargoToml.workspace.package) version;
+              name = "rio";
+              src = ./.;
+              cargoLock.lockFile = ./Cargo.lock;
+              buildInputs = runtimeDeps ++ buildDeps;
+              nativeBuildInputs = buildDeps;
+              buildNoDefaultFeatures = true;
+              buildFeatures = [ "x11" "wayland" ];
+              meta = {
+                  description = "A hardware-accelerated GPU terminal emulator focusing to run in desktops and browsers";
+                  homepage = "https://raphamorim.io/rio";
+                  license = lib.licenses.mit;
+                  platforms = lib.platforms.unix;
+                  changelog = "https://github.com/raphamorim/rio/blob/master/CHANGELOG.md";
+                  mainProgram = "rio";
+              };
+          };
 
-          meta = {
-            description = "A hardware-accelerated GPU terminal emulator powered by WebGPU";
-            homepage = "https://raphamorim.io/rio";
-            license = lib.licenses.mit;
-            platforms = lib.platforms.unix;
-            changelog = "https://github.com/raphamorim/rio/blob/master/CHANGELOG.md";
-            mainProgram = "rio";
+          mkDevShell = rust-toolchain: pkgs.mkShell {
+            shellHook = ''
+              export LD_LIBRARY_PATH="${pkgs.lib.makeLibraryPath (runtimeDeps ++ buildDeps)}:$LD_LIBRARY_PATH"
+            '';
```

```suggestion
              LD_LIBRARY_PATH="${pkgs.lib.makeLibraryPath (runtimeDeps ++ buildDeps)}:$LD_LIBRARY_PATH";
```
I wouldn't use a `shellHook` for this. Makes it easier to read in my opinion.

---

`shellHook` works fine for this anyways. But, as per your request, its done.